### ISSUE TITLE
ECO currently broken

### ISSUE THREAD

**Issue by Frisasky** [GameServerManagers/LinuxGSM#2549]
## User Story

users report the server did not start after installation

further investigation are needed

## Basic info

* Game: [Eco]

## Further Information

> [23:52:12.997]` [1] [ Info] [Eco] >>> START EcoServer, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null 
[23:52:13.023] [1] [ Info] [Eco] Eco Server 0.8.3.0 beta
[23:52:13.196] [1] [ Info] [Eco] Server Initialization...  
[23:52:13.198] [1] [ Info] [Eco] Starting LocalizationPlugin...  
[23:52:14.007] [1] [ Info] [Eco] Starting ModKitPlugin...  
[23:52:14.009] [1] [ Info] [Eco] Loading mods...  
[23:52:14.128] [1] [ Info] [Eco] Compiling mods...  
[23:52:18.382] [1] [ Info] [Eco] Mods recompiled with errors.
 

 

> at Eco.ModKit.RuntimeCompiler.HandleCompilerError (System.CodeDom.Compiler.CompilerResults results) [0x000b8] in <9f250128afe348fb9ec517685bc16590>:0 
at Eco.ModKit.RuntimeCompiler.CompileMods () [0x00180] in <9f250128afe348fb9ec517685bc16590>:0 
  at Eco.ModKit.RuntimeCompiler..ctor (System.String modsAssemblyPath) [0x00095] in <9f250128afe348fb9ec517685bc16590>:0 
  at Eco.ModKit.ModKitPlugin..ctor (System.Threading.Tasks.TaskCompletionSource 1[TResult] done) [0x00055] in <9f250128afe348fb9ec517685bc16590>:0 
  at (wrapper managed-to-native) System.Reflection.RuntimeConstructorInfo.InternalInvoke(System.Reflection.RuntimeConstructorInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeConstructorInfo.InternalInvoke (System.Object obj, System.Object[] parameters, System.Boolean wrapExceptions) [0x00005] in `<6ea21d94575948579c2890f43ad5bec5>:0

## To Reproduce

currently blank

## Expected behaviour

server running without any problem

**Comment by dgibbs64** [comment]
I have taken a look. I think that ECO is simply broken currently. I would get on to the developers

**Comment by Frisasky** [comment]
would be fixed in next release
https://github.com/StrangeLoopGames/EcoIssues/issues/13305

**Comment by DUDU54** [comment]
Eco doesn't not support Mono 6.x yet.
Maybe check the version of mono during the install ?

**Comment by Frisasky** [comment]
> 
> 
> Eco doesn't not support Mono 6.x yet.
> Maybe check the version of mono during the install ?

if you install mono from project repo theres no way to install older version

**Comment by DUDU54** [comment]
You can get it from release or archive

https://github.com/mono/mono/releases
https://download.mono-project.com/archive/

**Comment by redwyre** [comment]
Eco dev here. A few users have been bitten by this. Eco < 0.9.0 uses mono 5 (correctly stated on your own site https://linuxgsm.com/lgsm/ecoserver/). Don't use the mono repo, there is a breaking change in Mono 6 and Eco wont be updated before 0.9.0. From 0.8.3 there is an install.sh to install the packages from the correct OS repository. I would prefer you just run this but if you need to support older versions you can just copy the script.

FYI Eco 0.9.0+ doesn't use Mono and will drop install.sh if it's no longer needed.

**Comment by dgibbs64** [comment]
@redwyre Thank you for this update. I will use this info to address the problem.

**Comment by dgibbs64** [comment]
I have updated the website to state that mono6 does not work and that the next release of eco will resolve the problem. Also stated to make sure mono5 is installed.

I have decided not to make code changes to this as I feel it will not be worth it as when 0.9.0 is released the issue will be resolved and will no longer require mono anyway.

Holding issue until 0.9.0 is released to make dependency updates.
